pkg/controller/migmigration/task.go

```diff
@@ -465,8 +470,32 @@ func (t *Task) Run() error {
 		}
 		t.Requeue = NoReQ
 		t.next()
-	case InitialBackupFailed, FinalRestoreFailed:
+	case InitialBackupFailed, FinalRestoreFailed, ResourceValidationFailed:
 		t.next()
+	case ValidateResourcesMigrated:
+		migratedSuccessfully, err := t.validateStaticResourcesPresence()
```

Comment:
I'm not convinced this adds value.  It is essentially testing that velero backed up and restored all of the resources expected. Should this fail for one migration, our controller is broken and it will fail for all of them.  Seems this should be caught in e2e tests.

Reply:
Removed after short discussion. This check will make migration fail every time, and it should be a part of pre-migration health checks. @alexander-demichev We probably need to check if our migration tooling tag all the resources in the namespace, so that velero could pick it up. For example `ReplicaSet` created by an older version of the `Deployment` is left in the namespace, but is not picked by our migration. @jwmatthews Is this expected behaviour?

Reply:
You'll need to consider the owner references.  For example: for a Deployment <- ReplicaSet <- Pod  only the `Deployment` should be backed/Restored by velero since the Deployment will _spin up_ the ReplicaSet which will, in turn, _spin up_ the Pod.